### PR TITLE
fix: show add credit card modal only for cloud users

### PR DESCRIPTION
**Pull request by YounixM**

### CODE REVIEW

**Review by SagarRajput-7 — APPROVED** [review]

File: frontend/src/container/AppLayout/index.tsx [hunk 1]
```diff
@@ -71,7 +72,14 @@ function AppLayout(props: AppLayoutProps): JSX.Element {
 	const isPremiumChatSupportEnabled =
 		useFeatureFlags(FeatureKeys.PREMIUM_SUPPORT)?.active || false;
 
+	const isChatSupportEnabled =
+		useFeatureFlags(FeatureKeys.CHAT_SUPPORT)?.active || false;
+
+	const isCloudUserVal = isCloudUser();
+
 	const showAddCreditCardModal =
```
Comment (vikrantgupta25) [inline frontend/src/container/AppLayout/index.tsx:80]: will this hide intercom for signoz.io users ?

Comment (SagarRajput-7) [reply frontend/src/container/AppLayout/index.tsx:80]: Yes

Comment (vikrantgupta25) [inline frontend/src/container/AppLayout/index.tsx:80]: the same check is present at multiple places we should add everywhere ?

Comment (YounixM) [reply frontend/src/container/AppLayout/index.tsx:80]: This should be shown to only cloudUsers who don't have premium chat support enabled and who are in trail.

Comment (SagarRajput-7) [reply frontend/src/container/AppLayout/index.tsx:80]: Whats the expected behaviour here?

Comment (vikrantgupta25) [reply frontend/src/container/AppLayout/index.tsx:80]: not needed. resolving

**Review by vikrantgupta25 — APPROVED** [review]
lgtm

**Review by ahmadshaheer — APPROVED** [review]

**Review by SagarRajput-7 — APPROVED** [review]

**Review by makeavish — APPROVED** [review]

**Pull request merged**